dotCMS/src/main/java/com/dotcms/ai/client/AIClient.java

```diff
@@ -0,0 +1,108 @@
+package com.dotcms.ai.client;
+
+import com.dotcms.ai.domain.AIProvider;
+import com.dotcms.ai.domain.AIRequest;
+import com.dotcms.ai.domain.JSONObjectAIRequest;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpUriRequest;
+
+import javax.ws.rs.HttpMethod;
+import java.io.OutputStream;
+import java.io.Serializable;
+
+/**
+ * Interface representing an AI client capable of sending requests to an AI service.
+ *
+ * <p>
+ * This interface defines methods for obtaining the AI provider and sending requests
+ * to the AI service. Implementations of this interface should handle the specifics
+ * of interacting with the AI service, including request formatting and response handling.
+ * </p>
+ *
+ * <p>
+ * The interface also provides a NOOP implementation that throws an
+ * {@link UnsupportedOperationException} for all operations.
+ * </p>
+ *
+ * @author vico
+ */
+public interface AIClient {
+
+    AIClient NOOP = new AIClient() {
+        @Override
+        public AIProvider getProvider() {
+            return AIProvider.NONE;
+        }
+
+        @Override
+        public <T extends Serializable> void sendRequest(final AIRequest<T> request, final OutputStream output) {
+            throwUnsupported();
+        }
+
+        private void throwUnsupported() {
+            throw new UnsupportedOperationException("Noop client does not support sending requests");
+        }
+    };
+
+    /**
+     * Resolves the appropriate HTTP method for the given method name and URL.
+     *
+     * @param method the HTTP method name (e.g., "GET", "POST", "PUT", "DELETE", "patch")
+     * @param url the URL to which the request will be sent
+     * @return the corresponding {@link HttpUriRequest} for the given method and URL
+     */
+    static HttpUriRequest resolveMethod(final String method, final String url) {
+        switch(method) {
```

Comment:
I would use an strategy, here or wait for the white flag to use java 21 sintax; it could be reduce into a switch
for instance
https://belief-driven-design.com/looking-at-java-21-switch-pattern-matching-14648/

Reply:
I'm waiting for Java 21.

Reply:
wondering if patch and options should be here